x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AbstractThreadLocalRefCounted.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.elasticsearch.core.RefCounted;
+import org.elasticsearch.core.Releasable;
+
+/**
+ * Releasable, non-threadsafe version of {@link org.elasticsearch.core.AbstractRefCounted}.
+ * Calls to {@link AbstractThreadLocalRefCounted#decRef()} and {@link AbstractThreadLocalRefCounted#close()} are equivalent.
+ */
+abstract class AbstractThreadLocalRefCounted implements RefCounted, Releasable {
```

Comment:
Might call it `AbstractNonThreadSafeRefCounted` - `ThreadLocal` implies to me that it's a java `ThreadLocal` which has all kinds of fun baggage.